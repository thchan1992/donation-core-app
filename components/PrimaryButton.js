import { TouchableOpacity, Text, StyleSheet } from "react-native";
// PrimaryButton is a functional component that renders a button with a text and onPress functionality
const PrimaryButton = ({ text, onPress }) => {
  return (
    // TouchableOpacity is used to make the button responsive to touch/press events
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export default PrimaryButton;

const styles = StyleSheet.create({
  button: {
    width: 150,
    height: 70,
    margin: 10,
    borderRadius: 8,
    backgroundColor: "#28C7FA",
    borderRadius: 8,
    borderWidth: 1.5,
    borderColor: "#FF304F",
  },
  text: {
    alignSelf: "center",
    padding: 16,
    fontWeight: "bold",
    fontSize: 30,
  },
});
