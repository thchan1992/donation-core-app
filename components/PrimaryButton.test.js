import renderer from "react-test-renderer";
import React from "react";
import PrimaryButton from "./PrimaryButton";
import { TouchableOpacity, Text } from "react-native";

describe("PrimaryButton", () => {
  let onPress;

  beforeEach(() => {
    onPress = jest.fn();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<PrimaryButton text="Test Button" onPress={onPress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const component = renderer.create(
      <PrimaryButton text="Test Button" onPress={onPress} />
    );
    const instance = component.root;
    const button = instance.findByType(TouchableOpacity);
    button.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
