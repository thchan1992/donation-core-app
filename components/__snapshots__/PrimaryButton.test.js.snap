// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#28C7FA",
      "borderColor": "#FF304F",
      "borderRadius": 8,
      "borderWidth": 1.5,
      "height": 70,
      "margin": 10,
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "fontWeight": "bold",
        "padding": 16,
      }
    }
  >
    Test Button
  </Text>
</View>
`;
